Sources/Plasma/PubUtilLib/plGImage/plPNG.cpp

```diff
@@ -226,27 +229,72 @@ bool plPNG::IWrite(plMipmap* source, hsStream* outStream)
             row_ptrs[i] = (png_bytep)srcp + (i * stride);
         }
 
+        // Write image data
         png_write_image(png_ptr, row_ptrs);
+
+        // Prepare Textual Metadata
+        char time_string[26]; // RFC-1123: "Day, DD-Mon-YYYY hh:mm:ss";
+        time_t rawtime = time(nullptr);
+        strftime(time_string, sizeof(time_string), "%a, %d-%b-%Y %T", gmtime(&rawtime));
+        std::multimap<ST::string, ST::string> standard_fields;    
+        standard_fields.emplace("Software", plProduct::ProductString());
+        standard_fields.emplace("Creation Time", time_string);
+
+        png_text* text = new png_text[standard_fields.size() + textFields.size()];
+        auto text_buffer = new std::vector<char*>();
```

Comment:
This vector does not need to be on the heap.

Reply:
Yeah, especially since I forgot to keep it from leaking away at the end!